build: silence git describe error output

Biswa96 opened the pull request:
This change silent the warning from git describe command when
    the project is built using source tarball. The warning is
    fatal: not a git repository: 'neovim/.git'

dundargoc commented on cmake/GenerateVersion.cmake:
```diff
@@ -1,11 +1,13 @@
 execute_process(
   COMMAND git --git-dir=${NVIM_SOURCE_DIR}/.git --work-tree=${NVIM_SOURCE_DIR} describe --first-parent --dirty --always
   OUTPUT_VARIABLE GIT_TAG
   OUTPUT_STRIP_TRAILING_WHITESPACE
```
I'd prefer just slapping on a `ERROR_QUIET` to `execute_process` instead.

Biswa96 replied:
Would it be better to silent an error rather than fixing it?

dundargoc replied:
Your PR gives `v0.9.0-dev-` without a `.git` directory.

dundargoc replied:
I think so, yes. I don't think it's worth having accounting for every possible scenario.

Also, your PR gives `v0.9.0-dev-` without a `.git` repo, it should be `v0.9.0-dev`.

Biswa96 replied:
I have added the change as you suggested.

> your PR gives v0.9.0-dev- without a .git repo

It could be fixed by adding the same check with the `if (NVIM_VERSION_PRERELEASE)` condition. Or checking `GIT_TAG` after execute_process.

Biswa96 committed:
build: silent git describe error output

This change silent the warning from git describe command when
the project is built using source tarball. The warning is
fatal: not a git repository: 'neovim/.git'
